// @magidoc/plugin-svelte-marked/build/markdown/markedConfiguration.js
import { MarkdownBloquote, MarkdownBr, MarkdownCode, MarkdownCodeSpan, MarkdownDel, MarkdownDfn, MarkdownEm, MarkdownHeading, MarkdownHr, MarkdownHtml, MarkdownImage, MarkdownLink, MarkdownList, MarkdownListItem, MarkdownParagraph, MarkdownSpace, MarkdownStrong, MarkdownTable, MarkdownText, } from './components';
import Slugger from 'github-slugger';
import { Lexer } from 'marked';
export function parse(src) {
    const lexer = new Lexer();
    return lexer.lex(src);
}
export const defaultRenderers = () => ({
    heading: MarkdownHeading,
    blockquote: MarkdownBloquote,
    list: MarkdownList,
    list_item: MarkdownListItem,
    br: MarkdownBr,
    code: MarkdownCode,
    codespan: MarkdownCodeSpan,
    table: MarkdownTable,
    html: MarkdownHtml,
    paragraph: MarkdownParagraph,
    link: MarkdownLink,
    text: MarkdownText,
    def: MarkdownDfn,
    del: MarkdownDel,
    em: MarkdownEm,
    hr: MarkdownHr,
    strong: MarkdownStrong,
    image: MarkdownImage,
    space: MarkdownSpace,
    escape: MarkdownSpace,
});
export const defaultOptions = () => ({
    baseUrl: '/',
    slugger: new Slugger(),
});
