// mathlifier/dist/index.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }// src/math.ts
var _katex = require('katex'); var _katex2 = _interopRequireDefault(_katex);
function math(x, options) {
  options = { wrap: false, throwOnError: false, ...options };
  const xString = options.wrap ? x : `{${x}}`;
  return _katex2.default.renderToString(xString, options);
}

// src/display.ts

function display(x, options) {
  options = {
    overflowAuto: true,
    throwOnError: false,
    displayMode: true,
    ...options
  };
  let katexString = _katex2.default.renderToString(x, options);
  katexString = options.overflowAuto ? `<div style="overflow-x:auto;" class="mathlifier-display">${katexString}</div>` : katexString;
  return katexString;
}

// src/mathEnvironments.ts
function align(x, options) {
  options = {
    overflowAuto: true,
    throwOnError: false,
    displayMode: true,
    ...options
  };
  return displayEnvironment("align", x, options);
}
function alignStar(x, options) {
  options = {
    overflowAuto: true,
    throwOnError: false,
    displayMode: true,
    ...options
  };
  return displayEnvironment("align*", x, options);
}
function alignat(x, pairs, options) {
  const alignatOptions = {
    overflowAuto: true,
    throwOnError: false,
    displayMode: true,
    postBegin: `{${pairs}}`,
    ...options
  };
  return displayEnvironment("alignat", x, alignatOptions);
}
function alignatStar(x, pairs, options) {
  const alignatOptions = {
    overflowAuto: true,
    throwOnError: false,
    displayMode: true,
    postBegin: `{${pairs}}`,
    ...options
  };
  return displayEnvironment("alignat*", x, alignatOptions);
}
function gather(x, options) {
  options = {
    overflowAuto: true,
    throwOnError: false,
    displayMode: true,
    ...options
  };
  return displayEnvironment("gather", x, options);
}
function gatherStar(x, options) {
  options = {
    overflowAuto: true,
    throwOnError: false,
    displayMode: true,
    ...options
  };
  return displayEnvironment("gather*", x, options);
}
function equation(x, options) {
  options = {
    overflowAuto: true,
    throwOnError: false,
    displayMode: true,
    ...options
  };
  return displayEnvironment("equation", x, options);
}
function equationStar(x, options) {
  options = {
    overflowAuto: true,
    throwOnError: false,
    displayMode: true,
    ...options
  };
  return displayEnvironment("equation*", x, options);
}
var eqn = equation;
var eqnStar = equationStar;
function displayEnvironment(env, x, options) {
  options = {
    overflowAuto: true,
    throwOnError: false,
    displayMode: true,
    postBegin: "",
    ...options
  };
  return display(
    `\\begin{${env}}${options.postBegin}
    ${x}
  \\end{${env}}`,
    options
  );
}

// src/typesettingUtils.ts
var linebreak = "<br>";
var newline = "<br>";
function bold(x) {
  return `<strong>${x}</strong>`;
}
var strong = bold;
function em(x) {
  return `<em>${x}</em>`;
}
var emph = em;
var newParagraph = `<p>`;

// src/htmlTex.ts
function html(x) {
  return x;
}
function tex(x) {
  return "";
}
function htmlTex(x, y) {
  return x;
}























exports.align = align; exports.alignStar = alignStar; exports.alignat = alignat; exports.alignatStar = alignatStar; exports.bold = bold; exports.display = display; exports.em = em; exports.emph = emph; exports.eqn = eqn; exports.eqnStar = eqnStar; exports.equation = equation; exports.equationStar = equationStar; exports.gather = gather; exports.gatherStar = gatherStar; exports.html = html; exports.htmlTex = htmlTex; exports.linebreak = linebreak; exports.math = math; exports.newParagraph = newParagraph; exports.newline = newline; exports.strong = strong; exports.tex = tex;
